### ISSUE TITLE
IgnoredReturnValue false negative when annotation is on the class

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#3975]
## Expected Behavior of the rule
If I have this code:

```
org.assertj.core.api.Assertions.assertThat(somerhing)
```

`IgnoredReturnValue` should raise an issue because `org.assertj.core.api.Assertions` has the `@CheckReturnValue` annotation on it. But it doesn't.

The issue is that in the rule we just check for the annotation in the method but not in its class.

## Context
I spotted this issue working on #3973 and #3974

Related code:

- https://github.com/assertj/assertj-core/blob/main/src/main/java/org/assertj/core/api/Assertions.java#L146-L147
- https://github.com/assertj/assertj-core/blob/main/src/main/java/org/assertj/core/util/CheckReturnValue.java
- https://github.com/findbugsproject/findbugs/blob/master/findbugs/src/java/edu/umd/cs/findbugs/detect/BuildCheckReturnAnnotationDatabase.java#L120
